### DISCUSSION TITLE
github codespaces redirect issue

### DISCUSSION THREAD

**Discussion by RayBB** [community/community#4698] [Codespaces] [answered]
Hello,

This is kind of a complicated thing to explain so I won't go into the nitty gritty detail unless someone responds and says that this is an issue worth looking at.

Long story short:
* Open GitHub Codespaces with open library repo
* Run `docker-compose up`
* Use the "open in browser" feature to see the website hosted on port 8080 - everything looking good so far
* Most pages work but pages that have a redirect are somehow getting sent to localhost (for example if you click on a book on the homepage there is a redirect from /works/123 to /works/123/my_nice_title and that will force you to localhost. However, if you go to the exact end url where there is no redirect it works

Anyway, this _could_ be an application level issue. 
HOWEVER:
* The same issue does not happen on gitpod.io
* The same issue does not happen on local host
* A open library employee and myself looked and didn't see anything hardcoded or otherwise likely to cause this.

It's too bad because codespaces would be perfect for OpenLibrary volunteers who may have trouble running such a resource intense application locally.

Related issue:
https://github.com/internetarchive/openlibrary/issues/5438

**Comment by derekbekoe** [comment, marked as answer]
Hi @RayBB - Thanks for reporting this. We're taking a look at this scenario and should have more details in the next few days.

**Reply by RayBB** [reply]
@derekbekoe is this still being looked into?

**Reply by derekbekoe** [reply]
@RayBB This is still being looked into.
While not ideal as it requires a change to the application, here is a workaround as we continue to investigate a fix -  https://github.com/derekbekoe/openlibrary/commit/a1df5499264f5a0bc1e941daed7783e54392f5b8

**Reply by derekbekoe** [reply]
@RayBB This should now be fixed.

**Reply by RayBB** [reply]
It's working!  Thanks now :)

**Comment by ghost** [comment]
My system is bugged or hacked and every thing is not working properly please help guide me in what to do
